src/features/class-parsely-support.php

```diff
@@ -45,8 +45,9 @@ public function boot(): void {
 	 * @return array<number> Array of post IDs.
 	 */
 	public function add_parsely_trending_posts_query( array $posts, array $args ): array {
-		$parsely = $GLOBALS['parsely'];
-		if ( ! $parsely->api_secret_is_set() ) {
+		global $parsely;
+
+		if ( ! class_exists( '\Parsely\Parsely' ) || empty( $parsely ) || ! $parsely instanceof Parsely || ! $parsely->api_secret_is_set() ) {
```

Comment:
I think `$parsely instanceof Parsely` implies the first two checks.

```suggestion
		if ( ! $parsely instanceof Parsely || ! $parsely->api_secret_is_set() ) {
```

Reply:
this suggestion makes PHPStan angry. instead of ignoring the line, I added additional checks to appease.